client/connector.go

```diff
@@ -307,16 +340,24 @@ func (c *connector) prepareUpsertBatch(tableName string, columnNames []string, u
 					value = nil
 				}
 			}
+
 			// Set value to the last row.
-			if err = upsertBatchBuilder.SetValue(upsertBatchBuilder.NumRows-1, upsertBatchColumnIndex, value); err != nil {
-				upsertBatchBuilder.RemoveRow()
-				c.logger.With(
-					"name", "prepareUpsertBatch",
-					"error", err.Error(),
-					"table", tableName,
-					"columnID", columnID,
-					"value", value).Error("Failed to set value")
-				break
+			// compute hll value to insert
+			if column.HLLConfig.IsHLLColumn {
+				// here use original column data type to compute hll value
+				value, err = computeHLLValue(memCom.DataTypeFromString(column.Type), value)
+				if err != nil {
+					upsertBatchBuilder.RemoveRow()
+					c.logger.With("name", "prepareUpsertBatch", "error", err.Error(), "table", tableName, "columnID", columnID, "value", value).Error("Failed to set value")
+					break
+				}
+			} else {
```

Comment:
seems else branch is not correctly set the value?

Reply:
Done